l1-contracts/test/Decoder.t.sol

```diff
@@ -159,16 +159,22 @@ contract DecoderTest is Test {
     // K_LOGS_LEN = 4 + 8 = 12 (hex"0000000c")
     // I1_LOGS_LEN = 8 (hex"00000008")
     // I1_LOGS = 8 random bytes (hex"aafdc7aa93e78a70")
-    bytes memory emptyKernelData = hex"0000000c00000008aafdc7aa93e78a70";
-    (bytes32 logsHash, uint256 bytesAdvanced) = helper.computeKernelLogsHash(emptyKernelData);
-
-    // Note: First 32 bytes are 0 because those correspond to the hash of previous iteration and there was no previous
-    //       iteration.
-
-    bytes32 referenceLogsHash =
-      sha256(hex"0000000000000000000000000000000000000000000000000000000000000000aafdc7aa93e78a70");
+    bytes memory firstFunctionCallLogs = hex"aafdc7aa93e78a70";
+    // Prefix logs with length of kernel logs (12) and length of iteration 1 logs (8)
+    bytes memory encodedLogs = abi.encodePacked(hex"0000000c00000008", firstFunctionCallLogs);
+    (bytes32 logsHash, uint256 bytesAdvanced) = helper.computeKernelLogsHash(encodedLogs);
+
+    // Zero because this is the first iteration
+    bytes32 previousKernelPublicInputsLogsHash = bytes32(0);
+    bytes32 privateCircuitPublicInputsLogsHashFirstCall = sha256(firstFunctionCallLogs);
+
+    bytes32 referenceLogsHash = sha256(
+      abi.encodePacked(
+        previousKernelPublicInputsLogsHash, privateCircuitPublicInputsLogsHashFirstCall
+      )
+    );
 
-    assertEq(bytesAdvanced, emptyKernelData.length, "Advanced by an incorrect number of bytes");
+    assertEq(bytesAdvanced, encodedLogs.length, "Advanced by an incorrect number of bytes");
     assertEq(logsHash, referenceLogsHash, "Logs hash should be 0 when there are no logs");
```

Comment:
I guess the assertEq message is a copypasta from the previous test? :-P

Reply:
Addressed in d92e77f47e4d395b0daaf578d9fcf3040dedf29e . Thanks